File: WeeklyCode/week1/3Sum Closest/gaoqiang.py
class Solution:
    # @return an integer
    def threeSumClosest(self, num, target):
    	num.sort()
    	length=len(num)
    	if length<3:
    		return -1
    	minTarget=num[0]+num[1]+num[2]-target
    	for i in range(0,length-2):
    		j=i+1
    		k=length-1
    		while j<k:
    			tmp=num[i]+num[j]+num[k]-target
    			if tmp==0:
    				return target
    			if abs(tmp)<abs(minTarget):
    				minTarget=tmp
    			if tmp<0:
    				j=j+1
    			else:
    				k=k-1
    	return minTarget+target

if __name__=='__main__':
    s = Solution()
    print s.threeSumClosest([-1,2,1,-4],1)

File: WeeklyCode/week11/Search a 2D Matrix/gaoqiang.java
public class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null ||matrix.length==0||matrix[0].length==0){
            return false;
        }
        int m=matrix.length, n=matrix[0].length;
        int targetRow = -1;
        for(int i=0;i<m;i++){
            if(matrix[i][0]==target){
                return true;
            }
            if(matrix[i][0]>target){
                targetRow = i-1;
                break;
            }
            if(i==m-1){
                targetRow = i;
            }
        }
        if(targetRow<0){
            return false;
        }
        for(int i=0;i<n;i++){
            if(matrix[targetRow][i]==target){
                return true;
            }
        }
        return false;
    }
}